src/hotspot/share/opto/vectornode.cpp

```diff
@@ -304,6 +306,13 @@ bool VectorNode::is_vector_rotate_supported(int vopc, uint vlen, BasicType bt) {
     return true;
   }
 
+  // If target does not support variable shift operations then no point
+  // in creating a rotate vector node since it will not be disintegratable.
+  int has_scalar_shift = n->in(2)->Opcode() != Op_LoadI;
```

Comment:
Why is this decision not made on type but opcode? Are we sure that the Opcode will be Op_LoadI from all paths?

Reply:
Type will be an integer always since scalar rotation APIs accept only integral shits.

Reply:
This method enable SLP to validate if an operation is vectorizable, LoadI is a load from a memory and contiguous locations will eventually be packed as a LoadVector.

Reply:
The rotate by could come from an expression, it does not need to be a LoadI.

Reply:
Yes, pessimistic relaxed checking should be full proof here (only based on target feature check for variable shifts).

Reply:
Agree, let us be conservative and restrict based on target feature check for variable shifts.